libs/auth-api-lib/src/lib/entities/models/delegation-scope.model.ts

```diff
@@ -39,14 +39,42 @@ export class DelegationScope extends Model<DelegationScope> {
   @Column({
     type: DataType.STRING,
     primaryKey: true,
-    allowNull: false,
+    allowNull: true,
+    validate: {
+      oneButNotBoth() {
+        if (!(this.scopeName || this.identityResourceName)) {
+          return new Error(
+            'Either scopeName or identityResourceName must be specified.',
+          )
+        }
+        if (this.scopeName && this.identityResourceName) {
+          return new Error(
+            'ScopeName and identityResourceName can not both be specified.',
+          )
+        }
+      },
+    },
   })
   scopeName?: string
 
   @ForeignKey(() => IdentityResource)
   @Column({
     type: DataType.STRING,
     allowNull: true,
+    validate: {
+      oneButNotBoth() {
```

Comment:
You defined `oneButNotBoth` for this class... shouldn't this be using that instead?
```typescript
validate: {
  oneButNotBoth() {
    this.oneButNotBoth() // rename to this.eitherScopeNameOrIdentityResourceName()
  }
}
```